dm/dm/worker/subtask.go

```diff
@@ -897,24 +886,47 @@ func updateTaskMetric(task, sourceID string, stage pb.Stage, workerName string)
 }
 
 func (st *SubTask) GetValidatorError(errState pb.ValidateErrorState) []*pb.ValidationError {
-	st.RLock()
-	defer st.RUnlock()
-	if st.validator != nil && st.validator.Started() {
-		return st.validator.GetValidatorError(errState)
+	validator := st.getValidator()
+	if validator != nil && validator.Started() {
```

Comment:
I remember you mentioned we can get the status of the validator though it didn't start.